File: src/do02reen24/data-structure/programmers_42578.py
def solution(clothes):
    answer = 1
    clothesMap = dict()
    for c in clothes:
        if clothesMap.get(c[1]) == None:
            clothesMap[c[1]] = []
        clothesMap[c[1]].append(c[0])

    for value in clothesMap.values():
        answer *= (len(value) + 1)

    answer -= 1
    return answer

File: src/do02reen24/etc/programmers_42578.js
const getCombinationNumber = (object) => {
  let combination = 1;
  for (const key in object) {
    const itemLength = object[key].length + 1;
    combination *= itemLength;
  }
  return combination;
};

const solution = (clothes) => {
  const closet = {};
  clothes.forEach(([cloth, clothType]) => {
    closet[clothType]
      ? closet[clothType].push(cloth)
      : (closet[clothType] = [cloth]);
  });

  return getCombinationNumber(closet) - 1;
};
